service/src/test/java/bio/terra/workspace/db/ActivityLogDaoTest.java

```diff
@@ -0,0 +1,65 @@
+package bio.terra.workspace.db;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.db.model.ActivityLogChangedType;
+import java.util.UUID;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+public class ActivityLogDaoTest extends BaseUnitTest {
+
+  @Autowired ActivityLogDao activityLogDao;
+  @Autowired private NamedParameterJdbcTemplate jdbcTemplate;
+
+  @Test
+  public void setWorkspaceUpdatedDateAndGet() {
+    String workspaceId = UUID.randomUUID().toString();
+    assertNull(activityLogDao.getLastChangedDate(workspaceId));
+
+    activityLogDao.setChangedDate(workspaceId, ActivityLogChangedType.CREATE);
+
+    String changeType = getChangeType(workspaceId);
+    var latestDate = activityLogDao.getLastChangedDate(workspaceId);
+    assertNotNull(latestDate);
+    assertEquals(ActivityLogChangedType.CREATE.name(), changeType);
+  }
+
+  @Test
+  public void getLastUpdatedDate() {
```

Comment:
opt (optional) nit - I think first and second is enough, I don't think third adds anything.